ecommers/lib/ui/pages/add_shipping_address_page.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:ecommers/core/models/index.dart';
+import 'package:ecommers/core/provider_models/index.dart';
+import 'package:ecommers/shared/i18n.dart';
+import 'package:ecommers/ui/decorations/dimens/index.dart';
+import 'package:ecommers/ui/decorations/index.dart';
+import 'package:ecommers/ui/widgets/button/index.dart';
+import 'package:ecommers/ui/widgets/index.dart';
+import 'package:ecommers/ui/widgets/shipping_address_text_field.dart';
+import 'package:flutter/material.dart' hide BackButton;
+import 'package:flutter/services.dart';
+import 'package:provider/provider.dart';
+
+class AddShippingAddress extends StatefulWidget {
+  final ShippingAddressModel shippingAddres;
+  const AddShippingAddress({this.shippingAddres, Key key}) : super(key: key);
+
+  @override
+  _AddShippingAddressState createState() => _AddShippingAddressState();
+}
+
+class _AddShippingAddressState extends State<AddShippingAddress> {
+  TextEditingController _fullNameController,
+      _addressController,
+      _cityController,
+      _stateController,
+      _zipCodeController,
+      _countryController;
+  ShippingAddressProviderModel provider;
+
+  @override
+  void initState() {
+    _fullNameController =
+        TextEditingController(text: widget.shippingAddres?.fullName);
+    _addressController =
+        TextEditingController(text: widget.shippingAddres?.address);
+    _cityController = TextEditingController(text: widget.shippingAddres?.city);
+    _stateController =
+        TextEditingController(text: widget.shippingAddres?.state);
+    _zipCodeController =
+        TextEditingController(text: widget.shippingAddres?.zipCode);
+    _countryController =
+        TextEditingController(text: widget.shippingAddres?.country);
+
+    provider =
+        Provider.of<ShippingAddressProviderModel>(context, listen: false);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    disposeControllers();
+    provider.clean();
+    super.dispose();
+  }
+
+  void disposeControllers() {
+    _fullNameController.dispose();
+    _addressController.dispose();
+    _cityController.dispose();
+    _stateController.dispose();
+    _zipCodeController.dispose();
+    _countryController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.shippingAddres != null) {
+      provider?.shippingAddress = widget.shippingAddres;
+    }
+    final localization = I18n.of(context);
```

Comment:
please remove it and use without i18n reference, just use localization from dependencies